src/MCSolver.cpp

```diff
@@ -0,0 +1,545 @@
+/*
+ @file      MCSolver.cpp
+ @brief     utility functions for monte carlo neutron simulation
+ @author    Luke Eure
+ @date      January 12 2016
+*/
+
+#include "MCSolver.h"
+
+/*
+ @brief     constructor for MCSolver
+*/
+MCSolver::MCSolver(TrackGenerator* track_generator) : Solver(track_generator) {}
+
+/*
+ @brief     deconstructor for MCSolver
+*/
+MCSolver::~MCSolver() {}
+
+/*
+ @brief     set the geometry and root universe for the solver
+ @param     geometry the geometry of the solver
+*/
+void MCSolver::setGeometry(Geometry* geometry) {
+    _geometry = geometry;
+    _root_universe = geometry->getRootUniverse();
+}
+
+/*
+ @brief     set the flux for the solver
+ @param     geometry the flux of the solver
+*/
+void MCSolver::initializeFlux() {
+
+    _scalar_flux = new FP_PRECISION 
+        [_geometry->getNumFSRs()*_geometry->getNumEnergyGroups()];
+}
+    
+/*
+ @brief     generates and transports neutron histories, calculates the mean
+            crow distance
+ @param     n_histories number of neutron histories to run
+ @param     mat a Material object containing information
+            about the material
+ @param     flux a Flux object containing information about the flux
+ @param     num_batches the number of batches to be tested
+ @param     num_groups the number of neutron energy groups
+*/
+void MCSolver::computeEigenValue(int n_histories, int num_batches,
+        int num_groups) {
```

Comment:
We should be able to know the number of groups from the ``Geometry`` which was set during ``MCSolver::setGeometry(...)`` so we shouldn't have it as a parameter here